envdecode.go

```diff
@@ -131,61 +132,90 @@ func decode(target interface{}) (int, error) {
 		if env == "" {
 			env = defaultValue
 		}
-
 		if env == "" {
 			continue
 		}
 
 		setFieldCount++
 
-		switch f.Kind() {
-		case reflect.Bool:
-			v, err := strconv.ParseBool(env)
-			if err == nil {
-				f.SetBool(v)
-			}
+		if f.Kind() == reflect.Slice {
+			decodeSlice(&f, env)
+		} else {
+			decodePrimitiveType(&f, env)
+		}
+	}
 
-		case reflect.Float32, reflect.Float64:
-			bits := f.Type().Bits()
-			v, err := strconv.ParseFloat(env, bits)
-			if err == nil {
-				f.SetFloat(v)
-			}
+	return setFieldCount, nil
+}
 
-		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
-			if t := f.Type(); t.PkgPath() == "time" && t.Name() == "Duration" {
-				v, err := time.ParseDuration(env)
-				if err == nil {
-					f.SetInt(int64(v))
-				}
-			} else {
-				bits := f.Type().Bits()
-				v, err := strconv.ParseInt(env, 0, bits)
-				if err == nil {
-					f.SetInt(v)
-				}
+func decodeSlice(f *reflect.Value, env string) {
+	parts := strings.Split(env, ",")
+
+	values := parts[:0]
+	for _, x := range parts {
+		if x != "" {
```

Comment:
I wonder if we should also do `strings.TrimSpace` here.  It'd be unfortunate if your environment variable of `"1, 2"` failed to decode because of the extra space.